dist/pythonlibs/riotctrl_shell/cord_ep.py

```diff
@@ -0,0 +1,130 @@
+                                 r"\[[0-9a-f:]+(%\S+)?\](:\d+)?)"),
+            "epname": re.compile(r"ep name:\s+(?P<epname>.+)$"),
+            "ltime": re.compile(r"lifetime:\s+(?P<ltime>\d+)s$"),
+            "regif": re.compile(r"reg if:\s+(?P<regif>\S+)$"),
+            "location": re.compile(r"location:\s+(?P<location>\S+)$"),
+        }
+
+    def parse(self, cmd_output):
+        """
+        Parses output of CordEp::cord_ep_info() and CordEp::cord_ep_register()
+
+        >>> parser = CordEpRegistrationInfoParser()
+        >>> res = parser.parse("CoAP RD connection status:\\n"
+        ...                "RD address:"
+        ...                " coap://[fe80::48d5:9eff:fe98:6b74]:5683\\n"
+        ...                "   ep name: RIOT-760D2323760D2323\\n"
+        ...                "  lifetime: 60s\\n"
+        ...                "    reg if: /resourcedirectory\\n"
+        ...                "  location: /reg/1/")
+        >>> res['rdaddr']
+        'coap://[fe80::48d5:9eff:fe98:6b74]:5683'
+        >>> res['epname']
+        'RIOT-760D2323760D2323'
+        >>> res['ltime']
+        60
+        >>> res['regif']
+        '/resourcedirectory'
+        """
+        res = {}
+        for line in cmd_output.splitlines():
+            for key, comp in self.comps.items():
+                m = comp.search(line)
+                if m is not None:
+                    try:
+                        res[key] = int(m.group(key))
+                    except ValueError:
+                        res[key] = m.group(key)
+        return res if bool(res) else None
+
+
+class CordEpDiscoverParser(ShellInteractionParser):
+
+    def __init__(self):
+        self.comp = re.compile(r"the registration interface is"
+                               r"\s+'(?P<regif>\S+)'$")
+
+    def parse(self, cmd_output):
+        """
+        Parses output of CordEp::cord_ep_discover()
+
+        >>> parser = CordEpDiscoverParser()
+        >>> parser.parse(
+        ...     "the registration interface is '/resourcedirectory'")
+        '/resourcedirectory'
+        """
+        for line in cmd_output.splitlines():
+            m = self.comp.search(line)
+            if m is not None:
+                return m.group("regif")
+        return None
+
+
+# ==== ShellInteractions ====
+
+
+class CordEp(ShellInteraction):
+    REGISTER = "register"
+    DISCOVER = "discover"
+    UPDATE = "update"
+    REMOVE = "remove"
+    INFO = "info"
+
+    @staticmethod
+    def _make_addr(addr, port=None):
+        addr = f"[{addr}]"
+        if port is not None:
+            addr += ":{port:d}".format(port=port)
+        return addr
+
+    @ShellInteraction.check_term
+    def cord_ep_cmd(self, cmd, args=None, timeout=-1, async_=False):
+        cmd_str = "cord_ep {cmd}".format(cmd=cmd)
+        if args is not None:
+            cmd_str += " {args}".format(args=" ".join(str(a) for a in args))
+        return self.cmd(cmd_str, timeout=timeout, async_=async_)
```

Comment:
As in an error case the output either starts with `error:` or `usage:` for all commands, you could reflect that as a `RuntimeError`, as I did in `gnrc.py` and `netif.py`
```suggestion
        res = self.cmd(cmd_str, timeout=timeout, async_=async_)
        if ("error:" in res) or ("usage:" in res):
            raise RuntimeError(res)
        return res
```

Of course there should be a test case for each.

Reply:
Added error case for error and usage based on `test_txtsnd_error` in `test_netif.py`.